presto-hive/src/main/java/io/prestosql/plugin/hive/HiveUpdatablePageSource.java

```diff
@@ -62,13 +67,18 @@
     public static final int ORIGINAL_TRANSACTION_CHANNEL = 0;
     public static final int ROW_ID_CHANNEL = 1;
     public static final int BUCKET_CHANNEL = 2;
+    // Used for UPDATE operations
+    public static final int ROW_CHANNEL = 3;
     public static final int ACID_ROW_STRUCT_COLUMN_ID = 6;
     public static final Block DELETE_OPERATION_BLOCK = nativeValueToBlock(INTEGER, Long.valueOf(DELETE.getOperationNumber()));
+    public static final Block INSERT_OPERATION_BLOCK = nativeValueToBlock(INTEGER, Long.valueOf(INSERT.getOperationNumber()));
 
     // The bucketPath looks like .../delta_nnnnnnn_mmmmmmm_ssss/bucket_bbbbb(_aaaa)?
     public static final Pattern BUCKET_PATH_MATCHER = Pattern.compile("(?s)(?<rootDir>.*)/(?<dirStart>delta_[\\d]+_[\\d]+)_(?<statementId>[\\d]+)/(?<filenameBase>bucket_(?<bucketNumber>[\\d]+))(?<attemptId>_[\\d]+)?$");
     // The orignal file path looks like .../nnnnnnn_m(_copy_ccc)?
     public static final Pattern ORIGINAL_FILE_PATH_MATCHER = Pattern.compile("(?s)(?<rootDir>.*)/(?<filename>(?<bucketNumber>[\\d]+)_(?<rest>.*)?)$");
+    // After compaction, the bucketPath looks like .../base_nnnnnnn/bucket_bbbbb(_aaaa)?
```

Comment:
nit: s/compaction/major compaction/g
Minor compaction does not create the base directory

Reply:
Replaced.